distribution/lib/Standard/Base/0.0.0-dev/src/Warning.enso

```diff
@@ -106,7 +106,7 @@ type Warning
        > Example
          Detach warnings of a specific type.
 
-             result = Warning.detach_selected_warnings value (_.is_a Illegal_State_Error)
+             result = Warning.detach_selected_warnings value (_.is_a Illegal_State)
```

Comment:
```suggestion
             result = Warning.detach_selected_warnings value (_.is_a Illegal_State.Error)
```

While I'd expect `is_a` to work the way you expect too, I don't think that is the case currently.

I run an example check:
```
from Standard.Base import all

type Foo
    Bar x

main =
    y = Foo.Bar 42
    IO.println <| y.is_a Foo
    IO.println <| y.is_a Foo.Bar
```

and it gives back
```
False
True
```

We'd probably both expect both to be `True`, but currently that is not the case.

I think we need to revisit how `is_a` works, along with how `Panic.catch`, `Error.catch` and `Panic.recover` filtering also works.

cc: @hubertp @JaroslavTulach as I'm not sure who would be scheduling the fixes to these?

Reply:
This was broken before `Illegal_State_Error` is the type as well. I agree we should review `is_a` and `catch`.
Ideally would be able to pass both as you say.